tests/acceptance/comparative_booster_study/variables_lib.py

```diff
@@ -0,0 +1,140 @@
+import operator
+from functools import reduce
+
+from databuilder.codes import CTV3Code, ICD10Code
+from databuilder.query_language import case, when
+
+from . import schema
+
+
+def create_sequential_variables(
+    dataset, variable_name_template, events, column, num_variables, sort_column=None
+):
+    sort_column = sort_column or column
+    for index in range(num_variables):
+        next_event = events.sort_by(getattr(events, sort_column)).first_for_patient()
+        events = events.take(
+            getattr(events, sort_column) > getattr(next_event, sort_column)
+        )
+        variable_name = variable_name_template.format(n=index + 1)
+        setattr(dataset, variable_name, getattr(next_event, column))
```

Comment:
Yeah, it occurred to me while writing this that it might not be _loads_ of work to extend `first_for_patient()` to something like `first_n_for_patient(n)` giving you:
```
tests = covid_tests.first_n_for_patient(3)
dataset.test_date_0 = tests[0].date
dataset.test_date_1 = tests[1].date
dataset.test_date_1 = tests[2].date
```
Which has an obvious efficient implementation, unlike the approach above.

It's not a _lovely_ API, but maybe it's enough to solve some of the immediate problems without having to tackle the bigger questions of event-based outputs.

Reply:
Huh.

I think I'm going to claim that since this is _possible_, we'd be better off not polluting the API with workarounds for the lack of event-based outputs.

Reply:
I think I probably agree. Although they aren't _exactly_ equivalent in that the current workaround won't capture multiple events on the same day, which I think Will has pointed out before.

But overall yes, I think we're better off putting our efforts into proper event level support — which doesn't feel like it should be _that_ far away from being possible, does it?